sympy/geometry/parabola.py

```diff
@@ -407,21 +407,18 @@ def vertex(self):
             vertex = Point(focus.args[0], focus.args[1] - self.p_parameter)
 
         return vertex
-from sympy import *
-def tangent(a,b,g,f,h,c,x1,y1):  
-    x,y=symbols('x y')
-    P=Poly((x**2)*a+(y**2)*b+2*g*x+2*h*x*y+2*f*y+c,(x,y))
-    S=Eq((x**2)*a+(y**2)*b+(2*g*x)+(2*h*x*y)+(2*f*y),-c)
-    S1=P(x1,y1)
-    if S1==0:
-        print('Point lies on the curve')
-    elif S1<0:
-        print('Point lies inside the curve')
-    else:
-        print('Point lies outside the curve')
-    
-    T=Eq((a*x1+g+h*y1)*x+(b*y1+f+h*x1)*y,-(c+(g*x1)+(f*y1)))
-    p1=plot_implicit(S,show=False)
-    p2=plot_implicit(T,show=False)
-    p1.extend(p2)
-    p1.show()
+
+    def tangent(a, b, g, f, h, c, x1, y1):  
```

Comment:
See the issue page where I reference the existing `tangent_lines` and `perpendicular_lines` routines. What you have here is too specific; it would be appropriate to have a `tangent_lines` method for a `Curve` but the user would create the Curve and ask for the tangent lines, not pass parameters to create the `Curve`. But I wonder if it would make sense to have a `Conic` class that could be instantiated with an equation and would create the appropriate special case (Ellipse, Circle, Parabola or [Hyperbola](https://github.com/sympy/sympy/pull/17822)).